Framework/DataHandling/test/LoadEMUauTest.h

```diff
@@ -83,10 +83,66 @@ class LoadEMUauTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(logpm("DopplerFrequency"), 9.974, 1.0e-3);
     TS_ASSERT_DELTA(logpm("DopplerAmplitude"), 0.075, 1.0e-4);
 
+    TS_ASSERT_DELTA(logpm("ReactorPower"), 19.066, 1.0e-3);
+    TS_ASSERT_DELTA(logpm("ScanPeriod"), 240.733, 1.0e-3);
+    TS_ASSERT_DELTA(logpm("env_T3S1"), 0.0, 1.0e-3);
+
     // test some instrument parameters
     auto instr = output->getInstrument();
     auto iparam = [&instr](const std::string &tag) { return instr->getNumberParameter(tag)[0]; };
     TS_ASSERT_DELTA(iparam("AnalysedV2"), 630.866, 1.0e-3);
     TS_ASSERT_DELTA(iparam("SampleAnalyser"), 1.8, 1.0e-3);
   }
+
+  void test_load_beam_monitor() {
+    LoadEMUTar algToBeTested;
+
+    if (!algToBeTested.isInitialized())
+      algToBeTested.initialize();
+
+    std::string outputSpace = "LoadEMUauTest";
+    algToBeTested.setPropertyValue("OutputWorkspace", outputSpace);
+
+    // load with only beam monitor counts
+    std::string inputFile = "EMU0020493.tar";
+    algToBeTested.setPropertyValue("Filename", inputFile);
+    algToBeTested.setPropertyValue("SelectDetectorTubes", "16-50");
+    TS_ASSERT_THROWS_NOTHING(algToBeTested.execute());
+    TS_ASSERT(algToBeTested.isExecuted());
+
+    // get workspace generated and run
+    EventWorkspace_sptr output = AnalysisDataService::Instance().retrieveWS<EventWorkspace>(outputSpace);
+    auto run = output->run();
+
+    // useful lambda fns
+    auto logpm = [&run](const std::string &tag) {
+      return dynamic_cast<TimeSeriesProperty<double> *>(run.getProperty(tag))->firstValue();
+    };
+    auto logpmi = [&run](const std::string &tag) {
+      return dynamic_cast<TimeSeriesProperty<int> *>(run.getProperty(tag))->firstValue();
+    };
+    // check number of histograms and the sum
+    TS_ASSERT_EQUALS(output->getNumberHistograms(), 6592);
+    TS_ASSERT_EQUALS(output->getNumberEvents(), 3135);
+
+    // check the beam monitor rates
+    TS_ASSERT_DELTA(logpm("BeamMonitorRate"), 1482.76, 1.0e-2);
+    TS_ASSERT_DELTA(logpm("BeamMonitorBkgRate"), 1.01, 1.0e-2);
+    TS_ASSERT_EQUALS(logpmi("MonitorCounts"), 27510);
+
+    // repeat with the beam spectrum included
```

Comment:
Could this be refactored into a separate test? This seems to be doing quite a lot for a unit test.

Reply:
The test is specific to this instrument and I have added a detailed explanation are the start of the test.